zcash_client_sqlite/src/wallet/init/migrations/v_transactions_net.rs

```diff
@@ -0,0 +1,451 @@
+        init_wallet_db_internal(&mut db_data, None, &[add_transaction_views::MIGRATION_ID])
+            .unwrap();
+
+        // Create two accounts in the wallet.
+        let usk0 =
+            UnifiedSpendingKey::from_seed(&tests::network(), &[0u8; 32][..], AccountId::from(0))
+                .unwrap();
+        let ufvk0 = usk0.to_unified_full_viewing_key();
+        db_data
+            .conn
+            .execute(
+                "INSERT INTO accounts (account, ufvk) VALUES (0, ?)",
+                params![ufvk0.encode(&tests::network())],
+            )
+            .unwrap();
+
+        let usk1 =
+            UnifiedSpendingKey::from_seed(&tests::network(), &[1u8; 32][..], AccountId::from(1))
+                .unwrap();
+        let ufvk1 = usk1.to_unified_full_viewing_key();
+        db_data
+            .conn
+            .execute(
+                "INSERT INTO accounts (account, ufvk) VALUES (1, ?)",
+                params![ufvk1.encode(&tests::network())],
+            )
+            .unwrap();
+
+        // - Tx 0 contains two received notes of 2 and 5 zatoshis that are controlled by account 0.
+        // - Tx 1 creates two notes of 2 and 3 zatoshis for an external address, and a change note
+        //   of 2 zatoshis.
+        // - Tx 2 sends the entire change value from account 0 to account 1
+        db_data.conn.execute_batch(
+            "INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (0, 0, 0, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (0, 0, 'tx0');
+
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (0, 0, 0, '', 2, '', 'nf_a', false);
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (0, 3, 0, '', 5, '', 'nf_b', false);
+
+            INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (1, 1, 1, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (1, 1, 'tx1');
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_address, value)
+            VALUES (1, 2, 0, 0, '', 2);
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_address, value, memo)
+            VALUES (1, 2, 1, 0, '', 3, X'61');
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (1, 3, 0, '', 2, '', 'nf_c', true);
+
+            INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (2, 2, 2, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (2, 2, 'tx2');
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_address, value)
+            VALUES (2, 2, 0, 0, '', 2);
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (2, 1, 1, '', 2, '', 'nf_d', false);",
```

Comment:
I'm confused; the suggestion above just splits the received note into two smaller notes, but they both go to account 1. It also violates `CONSTRAINT tx_output UNIQUE (tx, output_index)`.

Reply:
The issue here is that `zcash_client_sqlite` is a publicly available library; we want to ensure that the semantics are consistent even for users who have wallets that *do* support multiple accounts.

There was definitely a bug here prior to these changes, but my contention remains that the notion of `sent transaction` and `received transaction` are leaky abstractions.

Reply:
I don't know what you mean by "leaky abstraction". But it's a notion that's on every wallet, centralized or decentralized, Fiat or Crypto, so we need to show what you received and what you sent. Can crypto be more complicated? yes for sure. But **we** as wallet SDK developers need to adapt to existing requirements regardless of what's possible protocol-wise to make sure that we have the same concepts as the other wallets of the world.

If Zcash can do more, that's great! but we want to focus on the 99,99% of users that just want to know what went out  and what came in.

Reply:
Actually wallets are specific interpretation of the blockchain data. So it would be expected that a tool made for building wallets has these specific domain specific abstractions to serve a certain scope. 

It can be the case that these views ended up being scoped too broadly and we need others that are tailored to mobile wallets. That's ok. Let's build them if so. 

If I shift my perspective and put my hat of a 9-to-5 wallet developer that has to deal with a lot of coins and other problems, I'd like the SDK to be straight forward and that does not require me to be giving it too much thought, meaning that I should be able to paint the screen with that info pretty much 1-to-1 without having to think about notes, etc.

Reply:
@pacu I think you're misunderstanding. The notion of "sent value" and "received value" are perfectly fine. The notions of "sent transaction" and "received transaction" are muddled, because a single transaction can be both.

Here's an example: you use your phone to scan to send a payment to a ZIP 321 QR code. One of the recipients is an address for some account in your wallet, either the one that you're sending the funds from or a different one.
I could, for example, in about 30 minutes build a trivially simple note management application that accepts an address, a balance, and a number of notes `n` and renders a QR code that splits up that balance in proportion to the first `n` numbers of the Fibonacci sequence. Enter the balance of the notes in your wallet, whatever `n` you want, scan the QR code and you have a transaction that spends all your notes and also where you receive the entire balance.